pkg/netflow/flowaggregator/flowaccumulator.go

```diff
@@ -16,62 +16,69 @@ import (
-type flowWrapper struct {
+// flowContext contains flow information and additional flush related data
+type flowContext struct {
 	flow                *common.Flow
 	nextFlush           time.Time
 	lastSuccessfulFlush time.Time
 }
 
 // flowAccumulator is used to accumulate aggregated flows
 type flowAccumulator struct {
-	flows             map[uint64]flowWrapper
+	flows             map[uint64]flowContext
 	mu                sync.Mutex
 	flowFlushInterval time.Duration
 	flowContextTTL    time.Duration
 }
 
-func newFlowWrapper(flow *common.Flow) flowWrapper {
+func newFlowContext(flow *common.Flow) flowContext {
 	now := timeNow()
-	return flowWrapper{
+	return flowContext{
 		flow:      flow,
 		nextFlush: now,
 	}
 }
 
-func newFlowAccumulator(aggregatorFlushInterval time.Duration) *flowAccumulator {
+func newFlowAccumulator(aggregatorFlushInterval time.Duration, aggregatorFlowContextTTL time.Duration) *flowAccumulator {
 	return &flowAccumulator{
-		flows:             make(map[uint64]flowWrapper),
+		flows:             make(map[uint64]flowContext),
 		flowFlushInterval: aggregatorFlushInterval,
-		flowContextTTL:    aggregatorFlushInterval * 5,
+		flowContextTTL:    aggregatorFlowContextTTL,
 	}
 }
 
 // flush will flush specific flow context (distinct hash) if nextFlush is reached
 // once a flow context is flushed nextFlush will be updated to the next flush time
-// Specific flow context in `flowAccumulator.flows` map will be deleted if `flowContextTTL`
+//
+// flowContextTTL:
+// flowContextTTL defines the duration we should keep a specific flowContext in `flowAccumulator.flows`
+// after `lastSuccessfulFlush`. // Flow context in `flowAccumulator.flows` map will be deleted if `flowContextTTL`
 // is reached to avoid keeping flow context that are not seen anymore.
+// We need to keep flowContext (contains `nextFlush` and `lastSuccessfulFlush`) after flush
+// to be able to flush at regular interval (`flowFlushInterval`).
+// Example, after a flush, flowContext will have a new nextFlush, that will be the next flush time for new flows being added.
 func (f *flowAccumulator) flush() []*common.Flow {
 	f.mu.Lock()
 	defer f.mu.Unlock()
 
 	var flowsToFlush []*common.Flow
-	for key, flow := range f.flows {
+	for key, flowCtx := range f.flows {
 		now := timeNow()
-		if flow.nextFlush.After(now) {
+		if flowCtx.nextFlush.After(now) {
 			continue
 		}
-		if flow.flow != nil {
-			flowsToFlush = append(flowsToFlush, flow.flow)
-			flow.lastSuccessfulFlush = now
-			flow.flow = nil
-		} else if flow.lastSuccessfulFlush.Add(f.flowContextTTL).Before(now) {
-			// delete flow wrapper if there is no successful flushes since `flowContextTTL`
+		if flowCtx.flow != nil {
+			flowsToFlush = append(flowsToFlush, flowCtx.flow)
+			flowCtx.lastSuccessfulFlush = now
+			flowCtx.flow = nil
+		} else if flowCtx.lastSuccessfulFlush.Add(f.flowContextTTL).Before(now) || flowCtx.lastSuccessfulFlush.Add(f.flowContextTTL).Equal(now) {
```

Comment:
Why do we need to check for equality here?

Reply:
When there is an equality, it means that the time between the last successful check and `now` is exactly the flowContextTTL.

If we don't delete the flow context, we will update the nextFlush to an additional flushInterval (e.g. 5min).
This means, the flowContext might require flowContextTTL (5min) + new flushInterval (5min) = 10min instead the expect 5min (define by flowContextTTL).

I noticed in practice, this can happen and when it happen, I think it's preferred to delete than to wait for nextFlush.

---

An alternative is to make this TTL expiration verification more often:
```
		} else if flowCtx.lastSuccessfulFlush.Add(f.flowContextTTL).Before(now) || flowCtx.lastSuccessfulFlush.Add(f.flowContextTTL).Equal(now) {
```

Currently, the TTL check is only on each nextFlush: 
```
		if flowCtx.nextFlush.After(now) {
			continue
		}
```

We can put the TTL check before nextFlush, but the downside is that we add some compute.

ReThinking about it, this alternative checks for TTL more often and might help lower memory usage by deleting the flows faster.

Reply:
@FlorianVeaux I implemented the alternative way described above here: https://github.com/DataDog/datadog-agent/pull/13226/commits/d8911834ea89fa29af7c93524d91f5b672be3bad

That way, we will check TTL at every 10s, instead of every flushInterval (e.g. 5min) for specific flow context.

Reply:
And remove equality check here https://github.com/DataDog/datadog-agent/pull/13226/commits/ebe2d28a0c39cf9c9b3c3a8e47bbea26b4531f69